### DISCUSSION TITLE
Add --process flag to monitoring or a process+monitor command

### DISCUSSION THREAD

**Discussion by Roboroads** [ptr727/PlexCleaner#157] [Ideas]
I would like to, when I start the monitor process, that it also processes the files that are already present in the monitored files. This way I don't have to spin up a container to process existing files + another monitoring changes.

Ideally I would spin u a container that goews over all the files, and then keeps running to also process new additions, so I can leave it open.

**Comment by ptr727** [comment]
Good idea, I added `--preprocess` to the `monitor` command (I prefer to not mix command and option names).  
Try it out in the develop pre-release branch.

Do note the constraints and issues on using [FileSystemWatcher](https://learn.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher?view=net-7.0) on Docker, NFS, SMB, etc.

I have personally only tested it on Windows directly, not in docker, and I know it does not work when the media is not on a local filesystem in Windows.

**Reply by Roboroads** [reply]
I had no problems with the monitor whilst watching a mounted filesystem on docker, if i moved some files that still worked. 

Thanks for adding! Now i can get rid of one of my containers 😁

**Comment by ptr727** [comment]
Released in v3.1.
